x/tx/signing/context.go

```diff
@@ -316,11 +317,26 @@ func (c *Context) ValidatorAddressCodec() address.Codec {
 	return c.validatorAddressCodec
 }
 
-// FileResolver returns the proto file resolver used by the context.
+// FileResolver returns the protobuf file resolver used by the context.
 func (c *Context) FileResolver() ProtoFileResolver {
 	return c.fileResolver
 }
 
+// TypeResolver returns the protobuf type resolver used by the context.
 func (c *Context) TypeResolver() protoregistry.MessageTypeResolver {
 	return c.typeResolver
 }
+
+// DefineCustomGetSigners defines a custom GetSigners function for a given
+// message type. It is defined as a function rather than a method on Context
+// because of how go generics work.
+//
+// NOTE: if a custom signers function is defined, the message type used to
+// define this function MUST be the concrete type passed to GetSigners,
+// otherwise a runtime type error will occur.
+func DefineCustomGetSigners[T proto.Message](ctx *Context, getSigners func(T) ([][]byte, error)) {
```

Comment:
I wonder if this should be done when constructing a context by setting options. We shouldn't really allow this after initialization

Reply:
I explored how to do this with depinject, with `CustomGetSignersFunc` as a `ManyPerContainerType`, but couldn't come up with anything too good.  Maybe we can change the API for `DefineCustomGetSigners`.  I think we'd need to drop the Generic and possibly pass the type name in.

Reply:
Yeah we can refactor the API. If that sounds like the right direction, let's first figure out how this works with the signing context options struct and then we can figure out depinject